### PR TITLE
Use non-plural name for 1 creature while viewing with Crystal Ball

### PR DESCRIPTION
**Pull request by gentooza**
fixing issue #3366 
taking an intermediate string variable, for checking monster count.
if == 1 is only one.

cheers!

### CODE REVIEW

File: src/fheroes2/dialog/dialog_quickinfo.cpp [hunk 1]
```diff
@@ -83,9 +83,9 @@ std::string ShowGuardiansInfo( const Maps::Tiles & tile, bool isOwned, bool exte
     if ( troop.isValid() && ( isOwned || ( extendedScoutingOption && scoutingLevel > Skill::Level::NONE ) ) ) {
         str.append( "\n \n" );
         str.append( _( "guarded by %{count} %{monster}" ) );
-
-        StringReplace( str, "%{monster}", StringLower( troop.GetMultiName() ) );
-        StringReplace( str, "%{count}", StringLower( Game::CountScoute( troop.GetCount(), isOwned ? static_cast<int>( Skill::Level::EXPERT ) : scoutingLevel ) ) );
+        std::string count = StringLower( Game::CountScoute( troop.GetCount(), isOwned ? static_cast<int>( Skill::Level::EXPERT ) : scoutingLevel ) );
```
Comment (ihhub) [inline src/fheroes2/dialog/dialog_quickinfo.cpp:86]: We shouldn't compare a string with 1 as this function might return "one" as well (for example in the future). The right approach would be to check scouting level and the amount of troops. If it's an expert level and the number is 1 then use non-plural name.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @gentooza , I left few comments here. Could you please take a look?

File: src/fheroes2/dialog/dialog_quickinfo.cpp (hunk 1 above)
Comment (gentooza) [reply src/fheroes2/dialog/dialog_quickinfo.cpp:86]: ok, cool
I'll handle it, ok?
cheers

Comment (ihhub) [reply src/fheroes2/dialog/dialog_quickinfo.cpp:86]: Hi @gentooza , what I meant is:
```
const int scoutingLevel = isOwned ? static_cast<int>( Skill::Level::EXPERT ) : scoutingLevel;
if ( scoutingLevel == Skill::Level::EXPERT && troop.GetCount() == 1 ) {
    StringReplace( str, "%{monster}", StringLower( troop.GetName() ) );
}
else {
    StringReplace( str, "%{monster}", StringLower( troop.GetMultiName() ) );
}

StringReplace( str, "%{count}", StringLower( Game::CountScoute( troop.GetCount(), scoutingLevel ) ) );
```

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @gentooza , would you mind to take a look at slightly enhanced solution being proposed? What do you think?

File: src/fheroes2/dialog/dialog_quickinfo.cpp (hunk 1 above)
Comment (gentooza) [reply src/fheroes2/dialog/dialog_quickinfo.cpp:86]: Hi! @ihhub 

yeah sure! I have only a question, if we do this:
```
if ( scoutingLevel == Skill::Level::EXPERT && troop.GetCount() == 1 ) {
[...]
```
we are going to ignore the *extendedScoutingOption* flag, isn't it?
same in ShowMonsterInfo function.

cheers

Comment (ihhub) [reply src/fheroes2/dialog/dialog_quickinfo.cpp:86]: According to the condition above this code `( isOwned || ( extendedScoutingOption && scoutingLevel > Skill::Level::NONE )` we can reach this code only is `isOwned` is set to **true** making `scoutingLevel` as **Skill::Level::EXPERT** or when `extendedScoutingOption` is set to true. So in this case there is no need to check for `extendedScoutingOption` again. Please correct me if I'm wrong.

**Comment by gentooza** [comment]
yes, totally agree
is it now ok?

**Comment by ihhub** [comment]
@gentooza , thank you very much for your contribution to the project!

**Comment by gentooza** [comment]
you are welcome!
and congrats, it's a very cool project

**Pull request merged**